### PR TITLE
Khwe 41 implement dark mode

### PR DESCRIPTION
**Pull request by WOLFRIEND**
image
image

### CODE REVIEW

File: src/components/DarkMode/DarkMode.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<div class="dark-mode">
```
Comment (WOLFRIEND) [inline src/components/DarkMode/DarkMode.html]: please use more descriptive name

File: src/components/DarkMode/DarkModeTheme.scss [hunk 2]
```diff
@@ -0,0 +1,218 @@
+.page_dark-mode {
```
Comment (WOLFRIEND) [inline src/components/DarkMode/DarkModeTheme.scss]: move to page.scss

File: src/components/Main/Main.js [hunk 3]
```diff
@@ -81,6 +82,14 @@ export class MainComponent {
     );
   }
 
+  async checkTheme() {
```
Comment (WOLFRIEND) [inline src/components/Main/Main.js]: move to dark mode switcher

File: src/components/Main/Main.js [hunk 4]
```diff
@@ -81,6 +82,14 @@ export class MainComponent {
...
+    const user = await AuthService.check();
+    const currentTheme = await MusicService.getTheme(user.uid);
```
Comment (WOLFRIEND) [inline src/components/Main/Main.js]: move getTheme to other service

File: src/components/Main/Main.js [hunk 5]
```diff
@@ -216,6 +225,7 @@ export class MainComponent {
 
   mount() {
     this.mountPoint.innerHTML = this.render();
+    this.checkTheme();
```
Comment (WOLFRIEND) [inline src/components/Main/Main.js]: move to dark mode switcher component mount

File: src/index.js [hunk 6]
```diff
@@ -5,3 +5,5 @@ const root = document.getElementById("root");
 
 const main = new MainComponent(root);
 main.mount();
+
+window.main = main;
```
Comment (WOLFRIEND) [inline src/index.js]: wat?

File: src/services/MusicService.js [hunk 7]
```diff
@@ -141,4 +141,15 @@ export class MusicService {
   static setNewRating(userId, songId, ratingValue) {
     return database.ref(`users/${userId}/rating/${songId}`).set(ratingValue);
   }
+
+  static getTheme(userId) {
```
Comment (WOLFRIEND) [inline src/services/MusicService.js]: move to other service

**Review by gamesminer — APPROVED** [review]

**Pull request merged**